tasks/undercloud.yml

```diff
@@ -146,3 +164,224 @@
     report_recommendations: '{{ undercloud_conf_enable_novajoin_recommendations }}'
   tags:
     - pre-undercloud-deploy
+
+- name: Set fact for novajoin user principal
+  set_fact:
+    novajoin_principal: "nova/{{ host_entry }}"
+  when: ipa_conf_stat.stat.exists
+
+- name: Set novajoin join.conf location
+  set_fact:
+    novajoin_conf_file: "{{ container_prefix }}/etc/novajoin/join.conf"
+
+- name: Verify that join.conf exists
+  stat:
+    path: "{{ novajoin_conf_file }}"
+  register: join_conf_st
+  tags:
+    - post-undercloud-deploy
+
+# We really can't continue without join.conf
+- name: Fail if join.conf is not present
+  fail:
+    msg: join.conf is necessary to continue the validations
+  when: not join_conf_st.stat.exists|bool
+  tags:
+    - post-undercloud-deploy
+
+# We need to lookup stuff from the novajoin join.conf file but its
+# only readable by root.  make a temporary copy here, and read everything
+# we need.
```

Comment:
can't we address this by adding become: true?

Reply:
No, it doesn't work - I tried.  Lookups always run in the context of the user running the playbook.